src/Combinatorics/Matroids/realization_space.jl

```diff
@@ -0,0 +1,989 @@
+The realizations are parametrized by
+[1   0   1   0   x2   x2                 x2^2    1    0]
+[0   1   1   0    1    1   -x1*x2 + x1 + x2^2    1    1]
+[0   0   0   1   x2   x1                x1*x2   x1   x2]
+in the Multivariate Polynomial Ring in x1, x2 over Rational Field
+avoiding the zero loci of the polynomials
+RingElem[x1 - x2, x2, x1, x2 - 1, x1 + x2^2 - x2, x1 - 1, x1*x2 - x1 - x2^2]
+
+julia> realization_space(uniform_matroid(3,6))
+The realizations are parametrized by
+[1   0   0   1    1    1]
+[0   1   0   1   x1   x3]
+[0   0   1   1   x2   x4]
+in the Multivariate Polynomial Ring in x1, x2, x3, x4 over Integer Ring
+avoiding the zero loci of the polynomials
+RingElem[x1*x4 - x2*x3, x2 - x4, x1 - x3, x1*x4 - x1 - x2*x3 + x2 + x3 - x4, x3 - x4, x4 - 1, x3 - 1, x3, x4, x1 - x2, x2 - 1, x1 - 1, x1, x2]
+
+```
+"""
+function realization_space(M::Matroid; B::Union{GroundsetType,Nothing} = nothing, 
+    F::AbstractAlgebra.Ring = ZZ, saturate::Bool=false, simplify::Bool=true,
+    char::Union{Int,Nothing}=nothing, q::Union{Int,Nothing}=nothing)::MatroidRealizationSpace
+
+    if char!=nothing && !isprime(char) && char!=0
+        error("The characteristic has to be 0 or a prime number.")
+    end
+
+    #Construct the base ring as F_p if q=p^k
+    if q!=nothing
+        isprimepower, p, k = is_prime_power_with_data(q)
+        if !isprimepower
+            error("The given q has to be a prime power.")
+        end
+        if char!=nothing && char!=p
+            error("The given characteristic doesn't match q.")
+        else
+            char = p
+        end
+    end
+
+    if char == 0
+        F = QQ
+    elseif char != nothing
+        F = GF(char)
+    end
+
+    rk = rank(M)
+    n = length(M)
+
+    goodM = isomorphic_matroid(M, [i for i in 1:n])
+
+    Bs = bases(goodM)
+    
+    if !isnothing(B) 
+        goodB = sort!(Int.([M.gs2num[j] for j in B]))
+    else
+        goodB = find_good_basis_heuristically(goodM)
+    end
+    polyR, mat = realization_space_matrix(goodM, goodB, F)
+
+    
+    eqs = Vector{RingElem}()
+    ineqs = Vector{RingElem}()
+
+    #need to catch the corner-case if there are no variables at all
+    if !(typeof(polyR) <: MPolyRing)
+        RS = MatroidRealizationSpace(ideal(polyR,0), ineqs, polyR, mat, F, char, q)
+        RS.realizable = true
+        return RS
+    end
+    
+    for col in subsets(Vector(1:n),rk)
+        
+        col_det = det(mat[:,col])
+        
+        if total_degree( col_det ) <= 0 
+            
+            if col_det != 0 && col in Bs 
+                if isunit( col_det ) 
+                    continue
+                end
+            elseif col_det != 0 # and col is not a basis
+                # determinant nonzero but set not a basis
+                push!( eqs, col_det )
+            elseif col in Bs 
+                #determinant zero but set is a basis, i.e. M is not realizable
+                return MatroidRealizationSpace(F, char, q)
```

Comment:
Nevermind, I had a look at the corresponding constructor and saw, that it sets the corresponding variable to false. Nevertheless, can there be matroid realization spaces without a defining ideal and equations? I thought this is exactly the information needed to compute realizability. I am very unsure whether I would allow these things to be `Nothing`.

Reply:
we removed that constructor.